doc/api/string_decoder.md

```diff
@@ -66,6 +66,16 @@ substitution characters appropriate for the character encoding.
 If the `buffer` argument is provided, one final call to `stringDecoder.write()`
 is performed before returning the remaining input.
 
+### stringDecoder.reset([encoding])
```

Comment:
I believe this is an implementation detail users don't have to know about.

Reply:
As it is, there is no way for the users to reset the string decoder, right? They have to create a new instance if needed. This will enable reusability.

Reply:
A new way to reset the state each time was not requested as far as I read the issue. So I also prefer not to expose the reset function. If I am correct the following should work.

```js
const { StringDecoder } = require('string_decoder');
const decoder = new StringDecoder('utf8');

decoder.write(Buffer.from([0xE2, 0x82])); // => ''
decoder.end(); // => '�'
decoder.write(Buffer.of(0x61)); // => 'a'
```